ansible_base/resource_registry/tasks/sync.py

```diff
@@ -0,0 +1,383 @@
+from __future__ import annotations  # support python<3.10
+
+import asyncio
+import csv
+import time
+from dataclasses import dataclass, field
+from datetime import timedelta
+from enum import Enum
+from io import StringIO, TextIOBase
+
+from asgiref.sync import sync_to_async
+from django.conf import settings
+from django.db import transaction
+from django.db.models import QuerySet
+from django.db.utils import DatabaseError, IntegrityError
+from django.utils import timezone
+from requests import HTTPError
+
+from ansible_base.resource_registry.models import Resource, ResourceType
+from ansible_base.resource_registry.rest_client import ResourceAPIClient, get_resource_server_client
+
+
+class ManifestNotFound(HTTPError):
+    """Raise when server returns 404 for a manifest"""
+
+
+class ResourceDeletionError(DatabaseError):
+    """Raise for deletion errors on Django ORM"""
+
+
+class ResourceSyncHTTPError(HTTPError):
+    """Custom catchall error"""
+
+
+ResourcePK = AnsibleID = ResourceHash = ServiceID = str
+
+
+class ResourceTypeName(str, Enum):
+    # keep the order
+    ORGANIZATION = "shared.organization"
+    TEAM = "shared.team"
+    USER = "shared.user"
```

Comment:
Hard-coding this certainly seems like a bad practice. There is a strong shared need to make use of logic inside of `ansible_base.resource_registry.apps`, like

```python
from ansible_base.resource_registry.registry import get_registry
[f'shared.{rt.model._meta.model_name}' for rt in registry.get_resources().values() if rt.managed_serializer]
```

This would give the same result as `[item.value for item in ResourceTypeName]` you use elsewhere... except maybe for maintaining the order. But if the order is necessary, existing data like `parent_resources` on the `ResourceConfig` model could be used.

Reply:
I implemented this way, however now it is processing `shared.authenticator` and I am not sure we want to sync that kind of resource, we can execute the command passing a specific list of resources to filter

```
django-admin resource_sync organization,team,user
```

Also when instantiating the SyncExecutor
```
SyncExecutor(resource_type_names=["shared.organization", "shared.team", "shared.user"]
```

c/c @newswangerd